HW8.py

```diff
@@ -0,0 +1,63 @@
+
+
+import sqlite3
+import csv
+import os
+import argparse
+
+DB_FILE = 'tasks_projects.db'
+
+
+def create_table_from_csv(csv_file_name, cur):
+#     print(csv_file_name)
+    table_name = csv_file_name.split('.')[0]  # Get table name from CSV file name
+#     print(table_name)
+    with open(csv_file_name, 'r') as f:
+        reader = csv.reader(f)
+#         print(reader)
+        headers = next(reader)  # first call of 'next' method reads first line containing columns names followed with data types, separated by colon (':')
+#         print(headers)
+        headers_with_types = ', '.join(headers).replace(':', ' ')  # make string like 'field1 varchar, field2 date' for insertion in CREATE statement
+        cur.execute('CREATE TABLE IF NOT EXISTS {}({})'.format(table_name, headers_with_types))
+
+        for row in reader:  # row is a list of values
+            qry = 'INSERT INTO {} VALUES ({})'.format(table_name,
+                                                      ', '.join(['"{}"'.format(val) for val in row])  # quotes are required for strings with spaces
+                                                      )
+#             print(qry)
+            cur.execute(qry)
```

Comment:
Please take a look at the"**executemany**" method.

A batch insert with executemany will be more efficient, and the performance difference will typically be quite large as your number of records gets large. There is a significant overhead to executing an insert statement, which you incur over and over again if you insert one row at a time.